apps/opt.c

```diff
@@ -326,6 +326,30 @@ int opt_int(const char *value, int *result)
     return 1;
 }
 
+static STRSTR_PAIR base[] = {
+    {"0x", "hex"},
+    {"0X", "hex"},
+    {"0", "oct"},
+    {NULL}
+};
+
+#define opt_number_error(v, b) \
```

Comment:
There is no reason for why it has to be a macro and not a static function. There is no reason for why base can't be declared static *inside* this function. There is no real reason to do a typedef, especially in another file. [In case one refers to typedefs chapter in coding style guide. There is no contradiction, because guide is talking about public and inter-module structures. It's kind of reflected in "final word" that effectively says "don't flash structures unless it's necessary". And it's not necessary to flash one that is used to describe data specific to a subroutine.]